torch/csrc/dynamo/guards.cpp

```diff
@@ -1009,6 +1009,12 @@ class LeafGuard {
   // This is on the hot path and avoids any refcounting code from pybind. This
   // is not exposed to Python and can only be called from C++.
   virtual bool check_nopybind(PyObject* value) = 0;
+  virtual bool check_nopybind(FrameLocalsMapping* map) {
+    throw std::runtime_error("fallback to python");
```

Comment:
Throwing here for now in order to make sure we don't fallback to Python during guard evaluation, giving a loud warning for us if we change guard implementation. We can allow for falling back to Python if we're concerned about breaking people.